modules/preview-react/pill/stories/Pill.stories.mdx

```diff
@@ -0,0 +1,237 @@
+
+### Removable Pills
+
+Removable Pills display an `X` icon after the label. They have a smaller, more specific focus state
+and click target to be more intentional about their actions and to avoid unintended removal of a
+Pill.
+
+You can define a removable Pill by providing a `variant='removable'` prop.
+
+```tsx
+<Pill variant="removable" onDelete={() => console.warn('clicked')}>
+  Pink Shirts
+  <Pill.IconButton />
+</Pill>
+```
+
+In this case, we use a `Pill.IconButton` because the `X` becomes the focusable and clickable
+element.
+
+The default icon for `Pill.IconButton` is `xSmallIcon` but this can also be overwritten by passing
+an `icon` prop to `Pill.IconButton`
+
+<ExampleCodeBlock code={RemovablePill} />
+
+### List of Pills
+
+Pills can often represent multiple pieces of information such as a filtered list of categories or
+skills.
+
+In order to achieve this, use our `HStack` component to wrap each `Pill` and space them out
+accordingly.
+
+<ExampleCodeBlock code={ListOfPills} />
+
+## Components
+
+### Pill
+
+This component is the container component for a Pill. It also provides a React Context model for its
+sub components. Based on the `variant` prop this component will render different styled Pills.
+
+#### Usage
+
+Example of read only:
+
+```tsx
+<Pill variant="readOnly">This is a read only</Pill>
+```
+
+If you give this component a `variant='default'` prop it will render as an interactive Pill which
+accepts subcomponents.
+
+```tsx
+<Pill variant="default" onClick={() => console.log('clicked')}>
+  <Pill.Avatar /> Regina Skeltor
+</Pill>
+```
+
+If you give this component a `variant='removable'` it will render as a span and you can have a
+removable Pill that has a smaller click target to prevent users from accidentally delete something.
+
+```tsx
+<Pill variant="removable" onDelete={() => console.log('handle remove')}>
+  Shoes
+  <Pill.IconButton />
+</Pill>
+```
+
+If you give the component an `onClick` and an `onDelete` and the `variant='default'` only `onClick`
+will be called and you'll see a warning in the console suggesting to either provide an onClick or
+change the variant to be `removable`.
+
+#### Props
+
+<ArgsTable of={Pill} />
+
+### Pill.Avatar
+
+This component renders an avatar. It supports all props of the `Avatar` component.
+
+#### Usage
+
+```tsx
+<Pill onDelete={() => console.log('handle remove')}>
+  <Pill.Avatar url={avatarUrl} />
+  Regina Skeltor
+  <Pill.IconButton />
+</Pill>
+```
+
+#### Props
+
+<ArgsTable of={Pill.Avatar} />
+
+### Pill.Count
+
+This component renders its `children` as the count.
+
+#### Usage
+
+```tsx
+<Pill onClick={() => console.warn('clicked')}>
+  Shoes
+  <Pill.Count>30</Pill.Count>
+</Pill>
+```
+
+#### Props
+
+<ArgsTable of={Pill.Count} />
+
+### Pill.Icon
+
+This component renders an `icon`. This component should only be used when `variant='default'`. By
+default this component render a `plusIcon` but it can be overriden.
```

Comment:
```suggestion
This component renders an `icon`. It not be used with the `default` styling – not `readOnly` or `removable` variants. By default it renders a `plusIcon` but it can be overridden by providing an icon to the `icon` prop.
```